src/dfa/dfa.rs

```diff
@@ -580,4 +580,80 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_int() -> Result<()> {
+        let mut parser = RegexParser::new();
+        let parsed_ast = parser.parse_into_ast(r"\-{0,1}\d+")?;
+
+        let mut nfa = NFA::new();
+        nfa.add_ast_to_nfa(&parsed_ast, NFA::START_STATE, NFA::ACCEPT_STATE)?;
+
+        let dfa = DFA::from_multiple_nfas(vec![nfa]);
+
+        assert_eq!(dfa.simulate("0"), (Some(0usize), true));
+        assert_eq!(dfa.simulate("1234"), (Some(0usize), true));
+        assert_eq!(dfa.simulate("-1234"), (Some(0usize), true));
+        assert_eq!(dfa.simulate("-0"), (Some(0usize), true));
+        assert_eq!(dfa.simulate("aba"), (None, false));
+        assert_eq!(dfa.simulate(""), (None, false));
+        assert_eq!(dfa.simulate("3.14"), (None, false));
+        assert_eq!(dfa.simulate("0.00"), (None, false));
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_float() -> Result<()> {
+        let mut parser = RegexParser::new();
+        let parsed_ast = parser.parse_into_ast(r"\-{0,1}[0-9]+\.\d+")?;
+
+        let mut nfa = NFA::new();
+        nfa.add_ast_to_nfa(&parsed_ast, NFA::START_STATE, NFA::ACCEPT_STATE)?;
+
+        let dfa = DFA::from_multiple_nfas(vec![nfa]);
+
+        assert_eq!(dfa.simulate("0.0"), (Some(0usize), true));
+        assert_eq!(dfa.simulate("-0.0"), (Some(0usize), true));
+        assert_eq!(dfa.simulate("-0.00001"), (Some(0usize), true));
+        assert_eq!(dfa.simulate("0.00001"), (Some(0usize), true));
+        assert_eq!(dfa.simulate("3.1415926"), (Some(0usize), true));
+        assert_eq!(dfa.simulate("-3.1415926"), (Some(0usize), true));
+
+        assert_eq!(dfa.simulate("0"), (None, false));
+        assert_eq!(dfa.simulate("1234"), (None, false));
+        assert_eq!(dfa.simulate("-1234"), (None, false));
+        assert_eq!(dfa.simulate("-0"), (None, false));
+        assert_eq!(dfa.simulate("aba"), (None, false));
+        assert_eq!(dfa.simulate(""), (None, false));
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_hex() -> Result<()> {
```

Comment:
This unit test failed. Can u help take a look @Louis-He? Here's the dumped NFA (manually drawing):
![image](https://github.com/user-attachments/assets/4feeeb67-0834-49f3-a28f-7f00b245abad)